src/Orleans.Core/Runtime/OutsideRuntimeClient.cs

```diff
@@ -19,6 +20,44 @@
 
 namespace Orleans
 {
+    [EventSource(Name="Microsoft-Orleans-OutsideRuntimeClientEvent")]
+    public class OrleansOutsideRuntimeClientEvent : EventSource
+    {
+        public static OrleansOutsideRuntimeClientEvent Log = new OrleansOutsideRuntimeClientEvent();
+        public void SendRequestStart()
+        {
+            if(EventSource.CurrentThreadActivityId!=Guid.Empty)
```

Comment:
I wonder if we should check for `this.Enabled()` instead?

Reply:
this.enabled will be checked by event source itself, https://referencesource.microsoft.com/#mscorlib/system/diagnostics/eventing/eventsource.cs,ae4fe1a0959bdb62

based on their source file